src/material.js

```diff
@@ -172,25 +172,67 @@ export class Material {
         return new Lattice(this.lattice);
     }
 
+    // returns the derived properties array for a material
+    getderivedProperties() {
+        return this.prop('derivedProperties', []);
+    }
+
+    /**
+     * @summary Returns the a specific derived property of a material, as defined by name
+     * @param {String} name
+     * @returns {Object}
+     */
+    getDerivedPropertyByName(name) {
+        return this.getDerivedProperties().find(x => x.name === name);
+    }
+
```

Comment:
The above functions then should be cleaned from webapp/material https://github.com/Exabyte-io/exabyte-web-app/pull/1676/files#diff-84e19e3fddd2ab3bc58b76c9b85a23a1774ef34b16c3fdcecb9fa74fd7c61489R94

Reply:
keeping `getDerivedProperties(flat=false)` function within the web-app per our discussion of the need to flatten the materials for display on the web-app.